zeppelin-server/src/main/java/org/apache/zeppelin/service/SessionManagerService.java

```diff
@@ -173,7 +173,7 @@ public SessionInfo getSessionInfo(String sessionId) throws Exception {
         } else {
           // if it is running before, but interpreterGroup is not running now, that means the session is stopped.
           // e.g. InterpreterProcess is terminated for whatever unexpected reason.
-          if (sessionInfo.getState().equals(SessionState.RUNNING.name())) {
+          if (sessionInfo.getState() != null && sessionInfo.getState().equals(SessionState.RUNNING.name())) {
```

Comment:
You could swap the objects of the 'equals'. A null check should then no longer be necessary.
```
SessionState.RUNNING.name().equals(sessionInfo.getState())
```

Reply:
Right, thanks for pointing this